src/cache/lru_evictor.cpp

```diff
@@ -58,13 +58,19 @@ bool LRUEvictor::LRUPartition::add_record(CacheRecord &record) {
 
 void LRUEvictor::LRUPartition::remove_record(CacheRecord &record) {
   std::unique_lock guard{m_list_guard};
+  if (!record.m_member_hook.is_linked()) {
```

Comment:
can you put some comment here, what does this if check and why should we return here?

Reply:
However why cache would remove a record that is not present, shouldn't we crash to catch the bug?

Reply:
I was thinking Debug assert and error log? It would not affect functionality in production, just a size mismatch

Reply:
This check is this is here for a reason right? Or is it just a defend check? If we are seeing this assert in our test, it means there is probably some race we don't know yet?

Reply:
This check was not there before, I added it because I hit the crash due to a bug in my implementation that was hit during the unit test. I added this now so that we will know where it failed

Reply:
This is just a defend check, we are not hitting it now in the tests